server/src/main/java/com/objectcomputing/checkins/services/memberSkill/MemberSkill.java

```diff
@@ -0,0 +1,100 @@
+package com.objectcomputing.checkins.services.memberSkill;
+
+import io.micronaut.data.annotation.AutoPopulated;
+import io.micronaut.data.annotation.TypeDef;
+import io.micronaut.data.model.DataType;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import java.util.Objects;
+import java.util.UUID;
+
+@Entity
+@Table(name ="member_skills")
+public class MemberSkill {
+
+    @Id
+    @Column(name="id")
+    @AutoPopulated
+    @TypeDef(type= DataType.STRING)
+    @Schema(description = "the id of the memberskill", required = true)
+    private UUID id;
+
+    @NotNull
+    @Column(name="memberid")
+    @TypeDef(type= DataType.STRING)
+    @Schema(description = "the id of the member profile", required = true)
+    private UUID memberid;
+
+    @NotNull
+    @Column(name="skillid")
+    @TypeDef(type= DataType.STRING)
+    @Schema(description = "the id of the skill", required = true)
+    private UUID skillid;
+
+    public MemberSkill() {
+    }
+
+    public MemberSkill(@NotNull UUID memberid, @NotNull UUID skillid) {
```

Comment:
Not nulls, aren't needed in the constructor since they are set at the field level.

Reply:
Good point. One could argue that it's expressive...but the fields aren't too far away either. ;-)

Reply:
👍

Reply:
Also for the purpose of just being expressive (and of use for IDEs) there is the @Nonnull annotation.

Reply:
And the micronaut team really likes the @NotBlank annotation

Reply:
You cant use that annotation for UUID though right?